R/wordbankr.R

```diff
@@ -4,10 +4,69 @@ utils::globalVariables(c(".", "n"))
 #' @importFrom rlang .data
 NULL
 
+#' Locate Wordbank Database
+#'
+#' @param mode A string indicating connection mode: one of \code{"local"},
+#'   \code{"remote"} (defaults to \code{"remote"}), or a date given in
+#'   \code{"YYYYMMDD"} or \code{"MMDDYYY"} format.
+#' @return A \code{src} object which is a connection to the Wordbank database.
+#' @keywords internal
+#'
+#' @examples
+#' \dontrun{
+#' db_info <- find_database("2017-03-22")
+#' }
+find_database <- function(mode = "remote") {
+  db_info = list()
+  if (mode == "remote") {
+    db_info[c("address", "dbname")] = c("server.wordbank.stanford.edu", "wordbank")
+  } else if (mode == "local") {
+    db_info[c("address", "dbname")] = c("localhost", "wordbank")
+  } else {
+    db_info["address"] <- "wordbank-backup.canyiscnpddk.us-west-2.rds.amazonaws.com"
+    mode_date <- tryCatch(lubridate::parse_date_time(mode, orders = c("mdY", "Ymd")),
+                          warning = function(e) {
+                            message("Error parsing date entered in mode variable.")
+                            message("Please enter dates as in 'Ymd' or 'mdY' format.")
```

Comment:
this is confusing to the user because the dates can actually be in a lot of different format as long as they're in that order